### PR TITLE
Prepare release r1.2

### LINKED ISSUE #72: Scope for RC.2 for Meta Release Fall 24
**Problem description**
0.2.0-rc.1 must be updated according to the comments from the Meta-Release reviewers
The scope RC.2 is the full alignment of the CFS API with the CAMARA Guidelines

**Possible evolution**
update the API according to the received comments

**Additional context**

- the release inherits the following scope: https://github.com/camaraproject/CallForwardingSignal/issues/49
- https://github.com/camaraproject/CallForwardingSignal/discussions/59#discussioncomment-10125696

### PR DESCRIPTION
**Pull request by hdamker**
#### What type of PR is this?

Add one of the following kinds:

* documentation
* subproject management

#### What this PR does / why we need it:

This is the release PR for r1.2. 

* Update of CHANGELOG.md
  * Moved new release to top
  * Applied the CHANGELOG.md template to r1.2
  * Replaced links to issues with links to PRs
  * Removed formal changes which are not relevant for API users

* Update Call_Forwarding_Signal.yaml
  * Removed comments from Design Guidelines (not meant to be in the YAML)
  * Updated x-camara-commonalities to the current available version
  * Updated version and server url for rc.2
  * Aligned server URL description with latest Commonalities guideline

* Deleted obsolete README.md in API_Definitions

* Cosmetic change within call-forwarding-signal-API-Readiness-Checklistst.md

#### Which issue(s) this PR fixes:

Fixes #72 #90 

#### Special notes for reviewers:

The PR should be merged only after all issues within scope of #72 are resolved (especially #90) and after the PR got an approval by one of @camaraproject/release-management_maintainers. The merge commit of this PR will get tagged as the release r1.2.

### CODE REVIEW

File: CHANGELOG.md
Comment (hdamker) [inline CHANGELOG.md:32]:
```suggestion
The pre-release is suitable for implementers, but it is not recommended to use the API with customers in productive environments.
```

**Comment by FabrizioMoggio** [comment]
@hdamker Just one question about this sentence:

"Special notes for reviewers:
The PR should be merged only after all issues within scope of https://github.com/camaraproject/CallForwardingSignal/issues/72 are resolved (especially https://github.com/camaraproject/CallForwardingSignal/issues/90) and after the PR got an approval by one of @camaraproject/release-management_maintainers. The merge commit of this PR will get tagged as the release r1.2."


and especially on this section:

"The merge commit of this PR will get tagged as the release r1.2."

My problem is that in this PR the version is still: wip. I can not create a Release Tag.

In my understanding the sequence is:

1. update your commit with the requested modifications (e.g. the Auth section text). The YAML still has version: wip
2. review and approve the PR by coworkers and release-management_maintainers. Still wip
3. merge the PR
4. create a new PR with just one modification, the version: v0.2.0-rc.2
5. approve the PR
6. merge the PR
7. create the Release Tag

is this sequence correct?

**Review by FabrizioMoggio — APPROVED** [review]
LGTM

File: CHANGELOG.md
Comment (hdamker) [inline CHANGELOG.md:43]:
```suggestion
* Improvement of user stories by @FabrizioMoggio in https://github.com/camaraproject/CallForwardingSignal/pull/86
* Aligned info.description Authorization and Authentication section with ICM 0.2.0 by @FabrizioMoggio in https://github.com/camaraproject/CallForwardingSignal/pull/91
```

**Comment by hdamker** [comment]
@FabrizioMoggio You have already seen by https://github.com/camaraproject/CallForwardingSignal/pull/91/commits/0b0b2f364d646e7230bbfe24f07691ba7dbf6c9c that I just missed to change the `version: wip`.

My intention was:

> In my understanding the sequence is:
> 
> 1. update your commit with the requested modifications (e.g. the Auth section text). The YAML still has version: wip

This change could have done also in a dedicated PR for the issue (which would have been merged before this one here). But I'm also fine with your commit https://github.com/camaraproject/CallForwardingSignal/pull/91/commits/3b72d16816dc5049802a0b3a1531c1a1df442b4f here. I updated the PR description ("Fixes ...") and the CHANGELOG.md.

> 2. review and approve the PR by coworkers and release-management_maintainers. Still wip

Yes, and this is the release PR, hence it does the change to v0.2.0-rc.2

> 3. merge the PR
~~> 4. create a new PR with just one modification, the version: v0.2.0-rc.2~~
~~> 5. approve the PR~~
~~> 6. merge the PR~~
> 7. create the Release Tag

**Review by FabrizioMoggio — APPROVED** [review]
OK

**Comment by FabrizioMoggio** [comment]
I think that everything is fixed. Waiting for a final approval by @camaraproject/release-management_maintainers before merging.

**Review by chinaunicomyangfan — APPROVED** [review]
LGTM

**Comment by hdamker** [comment]
> I think that everything is fixed. Waiting for a final approval by @camaraproject/release-management_maintainers before merging.

@FabrizioMoggio You are good to go ... I just asked @rartych for final view on the PR from release management perspective (as I didn't wanted to approve my own PR).

**Review by rartych — APPROVED** [review]

**Pull request merged**
